### PR TITLE
:mortar_board: Topic4 Clean up end sections wording

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Cleaned up some wording in the discussions on (a) nesting, (b) libraries, and (c) NumPy. 

### Why
The wording was either wrong, or wrong-ish, or arguable, or nuanced, or needs far more explaining, or bleh

### Where
Only the latter portion of the topic. There may be more cleanup needed for the earlier sections. It was just the latter sections that stood out particularly bad to me. 

### Additional Notes
Most of this, if not all, was vestigial content from back when the content was for UWO's COMP 2120. The focus of 2120 was a little different then a CS 101 course.

### CODE REVIEW

File: source/topic4.rst [hunk 1]
```diff
@@ -199,7 +199,7 @@ Nested conditionals
         else:
             print('Third Quadrant')
 			
-* In the above example, we *could* have done it without nesting and using just ``and`` and whatnot, but the nesting does make things easy to follow. 
+* In the above example, we *could* have done it without nesting by using ``and`` and whatnot, but for fun we can nest it
 
 * Again, no limit to how deep you nest... but mind the readability of your code!
```
Comment (twentylemon) [inline source/topic4.rst:204]: Python does actually have a limit to the amount of nesting, it's 20. [StackOverflow question](https://stackoverflow.com/questions/44972719/why-does-python-have-a-limit-on-the-number-of-static-blocks-that-can-be-nested) has some context.

**Review by twentylemon — APPROVED** [review]

File: source/topic4.rst (hunk 1 above)
Comment (jameshughes89) [reply source/topic4.rst:204]: Oh shit, I totally remember learning this earlier this year.

Comment (jdelamer) [reply source/topic4.rst:204]: Really? Good to know.

Comment (jameshughes89) [reply source/topic4.rst:204]: Neat, eh!?

**Pull request merged**